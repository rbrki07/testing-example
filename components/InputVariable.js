// @ts-check
import typedefs from "./../typedefs";
import React from "react";
import { StyleSheet, TextInput, View } from "react-native";
import { isNumeric } from "../helper/variables";

/**
 * @param {Object} params
 * @param {(variable: typedefs.VariableObject) => void} params.onSubmitEditing
 *
 * @returns {Object}
 */
const InputVariable = ({ onSubmitEditing }) => {
  return (
    <View>
      <TextInput
        placeholder={"Numeric variable"}
        onSubmitEditing={({ nativeEvent: { text } }) => {
          if (isNumeric({ text })) {
            onSubmitEditing({ variable: parseFloat(text) });
          }
        }}
        keyboardType={"numeric"}
        returnKeyType={"done"}
        clearButtonMode={"while-editing"}
        style={styles.input}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  input: {
    height: 48,
    width: 192,
    backgroundColor: "#f1faee",
    margin: 12,
    paddingLeft: 8,
    paddingRight: 8,
    borderRadius: 8,
    fontSize: 22,
    shadowOffset: {
      width: 1,
      height: 1,
    },
    shadowOpacity: 0.75,
    shadowRadius: 1,
    textAlign: "center",
  },
});

export { InputVariable };
