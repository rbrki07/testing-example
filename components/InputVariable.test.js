// @ts-check
import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import { InputVariable } from "./InputVariable";

describe("InputVariable component", () => {
  it("should call `onSubmitEditing` for numeric input", () => {
    const onSubmitEditing = jest.fn();
    const { getByPlaceholderText } = render(
      <InputVariable onSubmitEditing={onSubmitEditing} />
    );
    const input = getByPlaceholderText("Numeric variable");
    fireEvent(input, "submitEditing", { nativeEvent: { text: "4711" } });
    expect(onSubmitEditing).toBeCalledWith({ variable: 4711 });
  });

  it("should NOT call `onSubmitEditing` for nonnumeric input", () => {
    const onSubmitEditing = jest.fn();
    const { getByPlaceholderText } = render(
      <InputVariable onSubmitEditing={onSubmitEditing} />
    );
    const input = getByPlaceholderText("Numeric variable");
    fireEvent(input, "submitEditing", { nativeEvent: { text: "dhsh" } });
    expect(onSubmitEditing).not.toBeCalled();
  });
});
