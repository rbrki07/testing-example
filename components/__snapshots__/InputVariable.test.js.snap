// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputVariable component should render correctly 1`] = `
<View>
  <TextInput
    clearButtonMode="while-editing"
    keyboardType="numeric"
    onSubmitEditing={[Function]}
    placeholder="Numeric variable"
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#f1faee",
        "borderRadius": 8,
        "fontSize": 22,
        "height": 48,
        "margin": 12,
        "paddingLeft": 8,
        "paddingRight": 8,
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 0.75,
        "shadowRadius": 1,
        "textAlign": "center",
        "width": 192,
      }
    }
  />
</View>
`;
